### ISSUE TITLE
ifc -> gltf?

### ISSUE THREAD

**Issue by bertt** [hypar-io/Elements#401]
Hi, 

I'm converting a ifc 4 model (AC-20-Smiley-West-10-Bldg.ifc from this repository) to glTF using this code:

```
            var newModel = IFCModelExtensions.FromIFC("AC-20-Smiley-West-10-Bldg.ifc");
            newModel.ToGlTF("test.glb");
``` 
I'm using 0.8.0-alpha.9 of Hypar.Elements.Serialization.IFC. 

In the result it looks like there are missing colors and geometries. Is there something possible to improve the output?

Ifc model:

![image](https://user-images.githubusercontent.com/538812/95315819-4bcef080-0893-11eb-935a-89ec55e90b7e.png)

glTF:

![image](https://user-images.githubusercontent.com/538812/95315865-5ab5a300-0893-11eb-8149-842a3f851d9b.png)

**Comment by ikeough** [comment]
There is still much improvement needed on the ifc->gltf pipeline. There are a number of ways which color can be prescribed in IFC. We haven't settled on which one we want to support for colors and material support. You can read a post about this [here](https://forums.buildingsmart.org/t/why-is-it-so-difficult-to-get-colors-to-show-up/2312). Windows and doors are missing because we don't yet support boolean operations for openings. This will come soon as we add CSG support to our solids kernel.

**Comment by bertt** [comment]
ok good to know this is work in progress. About the geometry: I'm wondering why the ifc has 3 building levels, the glTF has 4?

**Comment by ikeough** [comment]
I believe the confusion is that in the gltf you're seeing the porch slab as if it were the porch _roof_. The porch roof and posts are missing, which is interesting because I believe we use this same model in our testing setup.

**Comment by ikeough** [comment]
The work we're doing in #406 get us here:
![image](https://user-images.githubusercontent.com/1139788/95889786-ba9ac680-0d37-11eb-870f-36806984b778.png)

**Comment by Moult** [comment]
Have you considered using the IFC to gltf workflow used by IfcOpenShell's IfcConvert?

**Comment by bertt** [comment]
@Moult  yes works very with IfcConvert, just investigating alternatives for more custom solutions.
